bin/git-create-branch

```diff
@@ -18,23 +26,29 @@ do
 	shift
 done
 
-if [[ -z $BRANCH ]] && [[ -n $REMOTE ]] 
+# handle ambiguous `-r` option argument by shift
+if [[ -z $BRANCH ]] && [[ -n $REMOTE ]]
 then
 	BRANCH=$REMOTE
 	REMOTE=origin
 fi
 
-test -z $BRANCH && echo "branch required." 1>&2 && exit 1
+test -z $BRANCH && echo "branch argument required." 1>&2 && exit 1
 if [[ -n $REMOTE ]]
 then
-	git ls-remote --exit-code $REMOTE &>/dev/null
+	git ls-remote --exit-code $REMOTE 1>/dev/null 2>/tmp/ls-remote-error
 	if [ $? = 0 ]
 	then
 		git push $REMOTE HEAD:refs/heads/$BRANCH
 		git fetch $REMOTE
 		git checkout --track -b $BRANCH $REMOTE/$BRANCH
 		exit $?
+	else
+		echo
+		echo "    Error connecting to remote '$REMOTE': $(</tmp/ls-remote-error)"
+		echo
 	fi
+	rm -f /tmp/ls-remote-error
```

Comment:
We need to preserve the exit code of `git checkout` or `git ls-remote`, and `exit $rc` here. Otherwise, the `$? = 0` branch will not remove the temp file and `$? != 0` branch will not exit correctly.

Reply:
good catch! thanks for the reviews. handled now.